communication/nats/dialog/dialog_waiter.go

```diff
@@ -70,10 +74,21 @@ func (waiter *dialogWaiter) ServeDialogs(dialogHandler communication.DialogHandl
 		if request.PeerID == "" {
 			return &responseInvalidIdentity, nil
 		}
+		registered, err := waiter.identityRegistry.IsRegistered(common.HexToAddress(request.PeerID))
+		if err != nil {
+			log.Error(waiterLogPrefix, "Peer identity registration check failed: ", err.Error())
+			return &responseInternalError, nil
+		}
+
+		if !registered {
+			//TODO should we tell peer that id is not registered (i.e. need to report more specific error?)
```

Comment:
Just log specific error as in line 93, and we dont respond with exact errors

Reply:
Will do